solidity.pegjs

```diff
@@ -664,13 +664,13 @@ LeftHandSideExpression
   / NewExpression
 
 Type
-  = literal:(Mapping / Identifier) members:("." Identifier)* parts:("[" __ (Expression)? __ "]")*
+  = literal:(Mapping / Identifier) members:("." Identifier)* parts:(__"[" __ (Expression)? __ "]")*

```

Comment:
I believe the __ can be added before parts: to avoid having to change indexes in the rule action.